### DISCUSSION TITLE
CTk Fullscreen window

### DISCUSSION THREAD

**Discussion by Cruzerskill** [TomSchimansky/CustomTkinter#1500] [Q&A]
I am working on a new project and this is my first attempt to use CTk. Is there a command available for fullscreen window on CTk that is similar to this one? upon looking for similar discussions, it seems that other commands for fullscreen window is not working as well. 
![for git](https://user-images.githubusercontent.com/46240643/233882558-2b092a74-c2c3-4a83-92c0-1257fb67a715.png)

**Comment by ohnishant** [comment]
customtkinter.CTk objects can be sent in fullscreen mode by changing the `-fullscreen` attribute to True. This is available in the official Tkinter docs
```py
root.attributes("-fullscreen", "True")
```

Here's a code snippet from a project I'm working on
```py
if __name__ == "__main__":
    app = App()
    app.title("Mega Mart Self Checkout System")
    app.attributes("-fullscreen", "True")
    app.columnconfigure(0, weight=1)
    app.rowconfigure(0, weight=1)
    app.mainloop()
```
![image](https://user-images.githubusercontent.com/52268717/233910068-83f3ca2e-0605-4c49-83b1-50c0ca6e23b2.png)

**Reply by Cruzerskill** [reply]
Yes, I've also tried to use this one and it's working but the idea that I want is when in full screen, the title bar will also reflect same as this one.
![for git](https://user-images.githubusercontent.com/46240643/233917255-115dc4b5-ef82-4d50-8450-5fc314087101.png)

**Comment by ohnishant** [comment]
Is it a maximized window that you are trying to get?

**Reply by Cruzerskill** [reply]
Yes, my goal is when the app is running using CTk, it will be in a maximized window.

**Comment by RickWalkerOne** [comment]
My code is working zoomed without problem, my code:

`
from customtkinter import CTk, set_default_color_theme
from libs.applibs.utils import *
from libs.uix.base import Base

class App(CTk):
    def __init__(self):
        self.palette = Palette()
        super().__init__()
        set_default_color_theme("assets/theme.json")
        self.screen_manager = ScreenManager()
        self.title("PyBAV")
        self.geometry("840x480")

        self.grid_rowconfigure(0, weight=1)
        self.grid_rowconfigure(1, weight=1)
        self.grid_rowconfigure(2, weight=1)
        self.grid_columnconfigure(0, weight=1)
        self.start()

        self.screen_manager.open_frames('base')

    def start(self):
        self.screen_manager.add_frame(Base(self), 'base')
        pass


if __name__ == "__main__":
    window = App()
    window.wm_iconbitmap(default='./assets/mfav/fav.ico')
    window.state("zoomed")
    window.mainloop()

`

it's my main.py

**Comment by mglenda** [comment]
When you want to start in maximized state then you can achieve it this way:

```
import customtkinter as ctk

w = ctk.CTk()
w._state_before_windows_set_titlebar_color = 'zoomed'
w.mainloop()
```

"w._state_before_windows_set_titlebar_color" attribute is not handled correctly inside _windows_set_titlebar_color(), which means that in mainloop() your predefined state will be forgotten unless you define the attribute manually

**Reply by eyedia** [reply]
Thanks, but this also disappears the windows task bar!

**Comment by rigvedmaanas** [comment]
Hi @Cruzerskill ,


You could try doing this if `root.state("zoomed")` doesn't work
```python
from customtkinter import *

root = CTk()

width = root.winfo_screenwidth()
height = root.winfo_screenheight()
geometry = str(width) + "x" + str(height)
root.geometry(geometry)



root.mainloop()
```
Hope it helps

**Comment by Ak0sTib0r** [comment]
Hello. I know this thread is a little old, but I was wondering of someone could help me. I tried the solutions here but for some reason the window is placed in the top left corner of my screen and only covers about 2/3-s of the height of my screen and about the same for the length. Here's my code. This the very beginning of a project.

```python
import customtkinter as ctk

ctk.set_appearance_mode("dark")
ctk.set_default_color_theme("dark-blue")

root = ctk.CTk()

rootWidth = root.winfo_screenwidth()
rootHeight = root.winfo_screenheight()
root.geometry(f"{rootWidth}x{rootHeight}")
root.title("editor")
root.wm_resizable(width = False, height = False)
root.attributes("-fullscreen", "True")

root.mainloop()
```

**Reply by Ak0sTib0r** [reply]
Forgot to mention, I tried it with resizable and non-resizable settings (line 12). Same result either way.

**Reply by CaiRookie-hub** [reply]
we‘re stuck the same page,bro :(
<img width="1280" height="764" alt="Snipaste_2026-07-18_15-36-12" src="https://github.com/user-attachments/assets/507d75d9-291b-412f-9f51-63085ff5e816" />

**Reply by CaiRookie-hub** [reply]
Hi bro. I've found a solution after 4 hours. Now you can use "CTk()._state_before_windows_set_titlebar_color = 'zoomed'" to solve this problem. And it won't hide your Windows taskbar! I think it's useful for you. 

Love from "THE EAST OF WORLD"

**Comment by Zyad205** [comment]
Wow I tagged this 3 years ago and it got labeled as a bug
